From 650843b05f71ef5e60e05afaf5636d20c9eb454c Mon Sep 17 00:00:00 2001
From: Nate Wiger <nate_wiger@playstation.sony.com>
Date: Mon, 19 May 2008 11:42:37 -0700
Subject: [PATCH] - Updated tzinfo to use Rational() instead of Rational.new! due to "private method new!" in Ruby 1.9
 - Added ^object_id$ as ignore pattern to fixing "redefining object_id" issue in Ruby 1.9

---
 .../associations/association_proxy.rb              |    2 +-
 activerecord/lib/active_record/named_scope.rb      |    4 +-
 .../vendor/tzinfo-0.3.8/tzinfo/offset_rationals.rb |  114 ++++++++++----------
 3 files changed, 60 insertions(+), 60 deletions(-)

diff --git a/activerecord/lib/active_record/associations/association_proxy.rb b/activerecord/lib/active_record/associations/association_proxy.rb
index 68503a3..ec16af3 100644
--- a/activerecord/lib/active_record/associations/association_proxy.rb
+++ b/activerecord/lib/active_record/associations/association_proxy.rb
@@ -49,7 +49,7 @@ module ActiveRecord
       alias_method :proxy_respond_to?, :respond_to?
       alias_method :proxy_extend, :extend
       delegate :to_param, :to => :proxy_target
-      instance_methods.each { |m| undef_method m unless m =~ /(^__|^nil\?$|^send$|proxy_)/ }
+      instance_methods.each { |m| undef_method m unless m =~ /(^__|^nil\?$|^send$|proxy_|^object_id$)/ }
 
       def initialize(owner, reflection)
         @owner, @reflection = owner, reflection
diff --git a/activerecord/lib/active_record/named_scope.rb b/activerecord/lib/active_record/named_scope.rb
index d43ebef..06edaed 100644
--- a/activerecord/lib/active_record/named_scope.rb
+++ b/activerecord/lib/active_record/named_scope.rb
@@ -102,7 +102,7 @@ module ActiveRecord
     
     class Scope
       attr_reader :proxy_scope, :proxy_options
-      [].methods.each { |m| delegate m, :to => :proxy_found unless m =~ /(^__|^nil\?|^send|class|extend|find|count|sum|average|maximum|minimum|paginate)/ }
+      [].methods.each { |m| delegate m, :to => :proxy_found unless m =~ /(^__|^nil\?|^send|^object_id$|class|extend|find|count|sum|average|maximum|minimum|paginate)/ }
       delegate :scopes, :with_scope, :to => :proxy_scope
 
       def initialize(proxy_scope, options, &block)
@@ -136,4 +136,4 @@ module ActiveRecord
       end
     end
   end
-end
\ No newline at end of file
+end
diff --git a/activesupport/lib/active_support/vendor/tzinfo-0.3.8/tzinfo/offset_rationals.rb b/activesupport/lib/active_support/vendor/tzinfo-0.3.8/tzinfo/offset_rationals.rb
index 5857de6..49ce415 100644
--- a/activesupport/lib/active_support/vendor/tzinfo-0.3.8/tzinfo/offset_rationals.rb
+++ b/activesupport/lib/active_support/vendor/tzinfo-0.3.8/tzinfo/offset_rationals.rb
@@ -27,63 +27,63 @@ module TZInfo
   # -14 and +14 hours to avoid having to call gcd at runtime.  
   module OffsetRationals #:nodoc:
     @@rational_cache = {
-      -50400 => Rational.new!(-7,12), 
-      -48600 => Rational.new!(-9,16),
-      -46800 => Rational.new!(-13,24),
-      -45000 => Rational.new!(-25,48),
-      -43200 => Rational.new!(-1,2),
-      -41400 => Rational.new!(-23,48),
-      -39600 => Rational.new!(-11,24),
-      -37800 => Rational.new!(-7,16),
-      -36000 => Rational.new!(-5,12),
-      -34200 => Rational.new!(-19,48),
-      -32400 => Rational.new!(-3,8),
-      -30600 => Rational.new!(-17,48),
-      -28800 => Rational.new!(-1,3),
-      -27000 => Rational.new!(-5,16),
-      -25200 => Rational.new!(-7,24),
-      -23400 => Rational.new!(-13,48),
-      -21600 => Rational.new!(-1,4),
-      -19800 => Rational.new!(-11,48),
-      -18000 => Rational.new!(-5,24),
-      -16200 => Rational.new!(-3,16),
-      -14400 => Rational.new!(-1,6),
-      -12600 => Rational.new!(-7,48),
-      -10800 => Rational.new!(-1,8),
-       -9000 => Rational.new!(-5,48),
-       -7200 => Rational.new!(-1,12),
-       -5400 => Rational.new!(-1,16),
-       -3600 => Rational.new!(-1,24),
-       -1800 => Rational.new!(-1,48),
-           0 => Rational.new!(0,1),
-        1800 => Rational.new!(1,48),
-        3600 => Rational.new!(1,24),
-        5400 => Rational.new!(1,16),
-        7200 => Rational.new!(1,12),
-        9000 => Rational.new!(5,48),
-       10800 => Rational.new!(1,8),
-       12600 => Rational.new!(7,48),
-       14400 => Rational.new!(1,6),
-       16200 => Rational.new!(3,16),
-       18000 => Rational.new!(5,24),
-       19800 => Rational.new!(11,48),
-       21600 => Rational.new!(1,4),
-       23400 => Rational.new!(13,48),
-       25200 => Rational.new!(7,24),
-       27000 => Rational.new!(5,16),
-       28800 => Rational.new!(1,3),
-       30600 => Rational.new!(17,48),
-       32400 => Rational.new!(3,8),
-       34200 => Rational.new!(19,48),
-       36000 => Rational.new!(5,12),
-       37800 => Rational.new!(7,16),
-       39600 => Rational.new!(11,24),
-       41400 => Rational.new!(23,48),
-       43200 => Rational.new!(1,2),
-       45000 => Rational.new!(25,48),
-       46800 => Rational.new!(13,24),  
-       48600 => Rational.new!(9,16),            
-       50400 => Rational.new!(7,12)}
+      -50400 => Rational(-7,12), 
+      -48600 => Rational(-9,16),
+      -46800 => Rational(-13,24),
+      -45000 => Rational(-25,48),
+      -43200 => Rational(-1,2),
+      -41400 => Rational(-23,48),
+      -39600 => Rational(-11,24),
+      -37800 => Rational(-7,16),
+      -36000 => Rational(-5,12),
+      -34200 => Rational(-19,48),
+      -32400 => Rational(-3,8),
+      -30600 => Rational(-17,48),
+      -28800 => Rational(-1,3),
+      -27000 => Rational(-5,16),
+      -25200 => Rational(-7,24),
+      -23400 => Rational(-13,48),
+      -21600 => Rational(-1,4),
+      -19800 => Rational(-11,48),
+      -18000 => Rational(-5,24),
+      -16200 => Rational(-3,16),
+      -14400 => Rational(-1,6),
+      -12600 => Rational(-7,48),
+      -10800 => Rational(-1,8),
+       -9000 => Rational(-5,48),
+       -7200 => Rational(-1,12),
+       -5400 => Rational(-1,16),
+       -3600 => Rational(-1,24),
+       -1800 => Rational(-1,48),
+           0 => Rational(0,1),
+        1800 => Rational(1,48),
+        3600 => Rational(1,24),
+        5400 => Rational(1,16),
+        7200 => Rational(1,12),
+        9000 => Rational(5,48),
+       10800 => Rational(1,8),
+       12600 => Rational(7,48),
+       14400 => Rational(1,6),
+       16200 => Rational(3,16),
+       18000 => Rational(5,24),
+       19800 => Rational(11,48),
+       21600 => Rational(1,4),
+       23400 => Rational(13,48),
+       25200 => Rational(7,24),
+       27000 => Rational(5,16),
+       28800 => Rational(1,3),
+       30600 => Rational(17,48),
+       32400 => Rational(3,8),
+       34200 => Rational(19,48),
+       36000 => Rational(5,12),
+       37800 => Rational(7,16),
+       39600 => Rational(11,24),
+       41400 => Rational(23,48),
+       43200 => Rational(1,2),
+       45000 => Rational(25,48),
+       46800 => Rational(13,24),  
+       48600 => Rational(9,16),            
+       50400 => Rational(7,12)}
     
     # Returns a Rational expressing the fraction of a day that offset in 
     # seconds represents (i.e. equivalent to Rational(offset, 86400)). 
-- 
1.5.5

